zio/bzngio/types.go

```diff
@@ -96,6 +96,7 @@ func (t *TypeFile) Dirty() bool {
 
 func ReadTypeContext(r io.Reader, zctx *resolver.Context) error {
 	reader := NewReader(r, zctx)
+	reader.zctx = zctx
```

Comment:
No, this is kind of hacky because the passed-in context gets binding to a `resolver.Mapper`. This would work but if the passed in reader doesn't have any records, the passed in Context doesn't get hydrated with the mapped types.